src/toil/batchSystems/kubernetes.py

```diff
@@ -496,47 +492,126 @@ def _getUpdatedBatchJobImmediately(self):
                     break
        
             
-        if jobObject is None:
-            # Say we couldn't find anything
-            return None
-
+            # Find a job that is done, failed, or stuck
+            jobObject = None
+            # Put 'done', 'failed', or 'stuck' here
+            chosenFor = ''
+            for j in self._ourJobObjects(onlySucceeded=True, limit=1):
+                # Look for succeeded jobs because that's the only filter Kubernetes has
+                jobObject = j
+                chosenFor = 'done'
+
+            if jobObject is None:
+                for j in self._ourJobObjects():
+                    # If there aren't any succeeded jobs, scan all jobs
+                    # See how many times each failed
+                    failCount = getattr(j.status, 'failed', 0)
+                    if failCount is None:
+                        # Make sure it is an int
+                        failCount = 0
+                    if failCount > 0:
+                        # Take the first failed one you find
+                        jobObject = j
+                        chosenFor = 'failed'
+                        break
+
+            if jobObject is None:
+                # If no jobs are failed, look for jobs with pods with
+                # containers stuck in Waiting with reason ImagePullBackOff
+                for j in self._ourJobObjects():
+                    pod = self._getPodForJob(j)
+
+                    if pod is None:
+                        # Skip jobs with no pod
+                        continue
+
+                    waitingInfo = getattr(pod.status.container_statuses[0].state, 'waiting')
+                    if waitingInfo is not None and waitingInfo.reason == 'ImagePullBackOff':
+                        # Assume it will never finish, even if the registry comes back or whatever.
+                        # We can get into this state when we send in a non-existent image.
+                        # See https://github.com/kubernetes/kubernetes/issues/58384
+                        jobObject = j
+                        chosenFor = 'stuck'
+                        logger.warning('Failing stuck job; did you try to run a non-existent Docker image?'
+                                       ' Check TOIL_APPLIANCE_SELF.')
+                        break
```

Comment:
You've copy-pasted and duplicated about half of the method at a deeper indentation level, before you get to setting up the watch. See https://github.com/DataBiosphere/toil/pull/2848/files#diff-183d11f3af830162fadf362f8c641fdfL499-R537

I don't think you want to do that. This probably resulted form a conflict, since I think I de-indented most of the method when I broke out `_getUpdatedBatchJobImmediately()`.